### PR TITLE
[tvos] Fix gn dependency problem for arm64 macbook

### PR DESCRIPTION
**Pull request by jasonzhangxx**
b/271481907
b/292107465

### CODE REVIEW

**Comment by andrewsavage1** [comment]
> b/271481907

Instead of reusing this bug could you create a new bug with the details of this issue please?

**Comment by jasonzhangxx** [comment]
> Also see b/272790873, we have this snippet https://github.com/youtube/cobalt/blob/main/starboard/build/config/BUILDCONFIG.gn#L90-L96 which changes the host toolchain manually for certain platforms. I think it could be confusing to have an arm64 toolchain setting a x64 cpu, so maybe it would be better to manually change the host cpu for arm64 platforms to be x64 instead, wdyt?

b/272790873 is empty, not sure what's the consideration behind that. I don't think manually replacing host cpu setting in starboard/build/config/BUILDCONFIG.gn  is much different than having x64  for arm64 toolchain setting. My understanding is BUILDCONFIG.gn is for all platforms. If that's true, I prefer to still set cpu to x64 in starboard/build/toolchain/mac/BUILD.gn, but adding a comment to explain why we use x64 there. So that, people can know that weird config is only for mac, and I think it's a little easier for other people to find why and how the toolchain cpu was set to x64.

**Comment by andrewsavage1** [comment]
The issue I have with setting the cpu in the toolchain like this is that this is not a host-specific toolchain, it's just marked as a general arm64 toolchain that incorrectly sets its own cpu. Plus, the host_cpu being arm64 in other places is probably incorrect, it should likely always be x64 to use the correct conditionals

**Comment by jasonzhangxx** [comment]
Ok, the new patch overwrite the host cpu in BUILDCONFIG.gn. Let me know if that aligns to your expectation.

File: starboard/build/config/BUILDCONFIG.gn [hunk 1]
```diff
@@ -90,6 +90,10 @@ import("//starboard/build/platform_path.gni")
 # TODO(b/272790873): Set host_cpu to x86 for 32 bit builds.
 if (target_cpu == "x86" || target_cpu == "arm") {
   _host_toolchain_cpu = "x86"
+} else
+# TODO(b/292107465): Set host_cpu to x64 for arm64 builds.
+if (target_cpu == "arm64") {
```
Comment (andrewsavage1) [inline starboard/build/config/BUILDCONFIG.gn:95]: Can you put this in a higher level else if instead of nesting it?

Comment (jasonzhangxx) [reply starboard/build/config/BUILDCONFIG.gn]: Discussed offline, moving the comment inside the if block, to make `} else if ()` in one line.

**Review by andrewsavage1 — APPROVED** [review]

**Pull request merged**